editor/scene/3d/node_3d_editor_plugin.cpp

```diff
@@ -8043,6 +8043,15 @@ void Node3DEditor::_selection_changed() {
 		selected->update_gizmos();
 		selected = nullptr;
 	}
+
+	const List<Node *> &top_selected = editor_selection->get_top_selected_node_list();
+	if (top_selected.size() == 1) {
+		Node3D *new_selected = Object::cast_to<Node3D>(top_selected.back()->get());
+		if (new_selected != selected) {
+			edit(new_selected);
+		}
+	}
```

Comment:
It might be worth adding a comment above the added block to explain why it's there:

```suggestion
	// Ensure gizmo updates are performed when the selection changes
	// outside of the 3D view (see GH-106713).
	const List<Node *> &top_selected = editor_selection->get_top_selected_node_list();
	if (top_selected.size() == 1) {
		Node3D *new_selected = Object::cast_to<Node3D>(top_selected.back()->get());
		if (new_selected != selected) {
			edit(new_selected);
		}
	}
```